### ISSUE TITLE
ArgumentCountError: Too few arguments to function

### ISSUE THREAD

**Issue by willdashwood** [openprovider/OP-WHMCS7#40]
When updating contact information on certain domains the following error is generated:

```
ArgumentCountError: Too few arguments to function OpenProvider\WhmcsRegistrar\Library\Handle::update(), 0 passed in /home/nginx/domains/DOMAIN.com/public/billing/modules/registrars/openprovider/Library/Handle.php on line 147 and exactly 1 expected in /home/nginx/domains/DOMAIN.com/public/billing/modules/registrars/openprovider/Library/Handle.php:270
Stack trace:
#0 /home/nginx/domains/DOMAIN.com/public/billing/modules/registrars/openprovider/Library/Handle.php(147): OpenProvider\WhmcsRegistrar\Library\Handle->update()
#1 /home/nginx/domains/DOMAIN.com/public/billing/modules/registrars/openprovider/openprovider.php(651): OpenProvider\WhmcsRegistrar\Library\Handle->updateOrCreate(Array, 'registrant')
#2 /home/nginx/domains/DOMAIN.com/public/billing/vendor/whmcs/whmcs-foundation/lib/Module/AbstractModule.php(0): openprovider_SaveContactDetails(Array)
#3 /home/nginx/domains/DOMAIN.com/public/billing/vendor/whmcs/whmcs-foundation/lib/Module/Registrar.php(0): WHMCS\Module\AbstractModule->call('SaveContactDeta...', Array)
#4 /home/nginx/domains/DOMAIN.com/public/billing/vendor/whmcs/whmcs-foundation/lib/Domains.php(0): WHMCS\Module\Registrar->call('SaveContactDeta...', Array)
#5 /home/nginx/domains/DOMAIN.com/public/billing/vendor/whmcs/whmcs-foundation/lib/Domains.php(0): WHMCS\Domains->moduleCall('SaveContactDeta...', Array)
#6 /home/nginx/domains/DOMAIN.com/public/billing/whmcsadmin/clientsdomaincontacts.php(0): WHMCS\Domains->saveContactDetails(Object(WHMCS\Client), Array, Array, Array)
#7 {main}
```

I'm not sure under what conditions this occurs but I can provide a list of domains in private that this occurs with if that helps. This could be related to issue #23.

**Comment by soberoctober** [comment]
hello @willdashwood 

what update are you trying to push?
can you submit a diff in a ticket to support@openprovider.nl ?

**Comment by willdashwood** [comment]
I thought I'd raised a ticket for the domains that caused the error but I think I'm confusing that with the issue #28. I haven't personally noticed the error since. I'll try and find out more information.

**Comment by willdashwood** [comment]
@soberoctober I've had the issue pop up again so I'm dropping you an email with the domain name in question.

**Comment by tamerciaga** [comment]
Is there any news about this? We are getting this same issue with WHMCS 7.6.2, Apache and PHP7.

**Comment by soberoctober** [comment]
@willdashwood @tamerciaga please try applying the following fix in `modules/registrars/openprovider/src/Handle.php`

```
@@ -157,7 +157,7 @@ class Handle
        if($action == 'create')
            $this->findOrCreate($params, $type);
        else
            $this->update();

        return $this->model->handle;
    }
```

```
@@ -157,7 +157,7 @@ class Handle
        if($action == 'create')
            $this->findOrCreate($params, $type);
        else
 $this->update($params);

        return $this->model->handle;
    }
```

**Comment by soberoctober** [comment]
fixed in [v3.0](https://github.com/openprovider/OP-WHMCS7/tree/v3.0)
